src/backend/distributed/commands/index.c

```diff
@@ -522,6 +523,49 @@ GetCreateIndexRelationLockMode(IndexStmt *createIndexStatement)
 }
 
 
+/*
+ * ReindexStmtFindIndexOid returns the oid of the relation on which the index exist in reindex stmt.
```

Comment:
The comment is missing an important thing: We are also acquiring the relevant lock